homeassistant/components/lightwave/__init__.py

```diff
@@ -6,24 +6,38 @@
 import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.discovery import async_load_platform
 
-LIGHTWAVE_LINK = "lightwave_link"
-
 DOMAIN = "lightwave"
+LIGHTWAVE_LINK = "lightwave_link"
+LIGHTWAVE_TRV_PROXY = "lightwave_proxy"
+LIGHTWAVE_TRV_PROXY_PORT = "lightwave_proxy_port"
+PROXY_IP = "trv_proxy_ip"
+PROXY_PORT = "trv_proxy_port"
 
 
 CONFIG_SCHEMA = vol.Schema(
     {
         DOMAIN: vol.Schema(
             vol.All(
-                cv.has_at_least_one_key(CONF_LIGHTS, CONF_SWITCHES),
+                cv.has_at_least_one_key(CONF_LIGHTS, CONF_SWITCHES, "trv"),
                 {
+                    vol.Required(CONF_HOST): cv.string,
+                    vol.Optional(PROXY_IP): cv.string,
+                    vol.Optional(PROXY_PORT): cv.string,
```

Comment:
If `PROXY_IP` and `PROXY_PORT` are only used if a trv is specified moved them into trv config section.